File: python/2nd_week/Praktikum2/latihanpraktikum2_python/Hardware.py
from Product import Product

# implementasi class Hardware dengan parent product
class Hardware(Product):
    # constructor
    def __init__(self):
        self.brand = ""
        self.model = ""
    
    # set and get method
    def setBrand(self, brand):
        self.brand = brand
    
    def setModel(self, model):
        self.model = model
    
    def getBrand(self):
        return self.brand
    
    def getModel(self):
        return self.model

File: cpp/2nd_Week/Praktikum2/latihanpraktikum2_cpp/Hardware.cpp
#include "Product.cpp"

// child class dengan parent class Product
//implementasi class hardware
class Hardware: public Product
{
    private:
        string brand, model;        //atribut private dengan tipe data string       
    public:
        // constructor
        Hardware() {}

        /* set and get method */
        void setBrand(string paramBrand) {      //set brand method
            brand = paramBrand;
        }

        void setModel(string paramModel) {      //set model method
            model = paramModel;
        }

        string getBrand() {                 //get brand method
            return brand;
        }

        string getModel() {                 // get model method
            return model;
        }

        // destructor
        ~Hardware() {}
};